src/iab.ts

```diff
@@ -91,6 +91,97 @@ export const TcfV2VendorList = t.intersection([
 ]);
 
 /**
- * Overload StoredContentfulEntry as a type
+ * Overload TcfV2VendorList as a type
  */
 export type TcfV2VendorList = t.TypeOf<typeof TcfV2VendorList>;
+
+/**
+ * TCF GVL v3 purpose configuration
+ */
+export const TcfGvlV3Purpose = t.type({
+  id: t.number,
+  name: t.string,
+  description: t.string,
```

Comment:
Would it make sense for TcfGvlV3Purpose to inherit these from TcfGvlV3DataCategory or am I trying too hard to be DRY? (I'm honestly not sure.)

Reply:
I like the idea of being more DRY but I was more focused on the transition to the new version. We have to totally transition to v3 of the vendors list by Sept 30, so I was thinking about totally removing the v2 types after we make the transition, unless you can think of any other reason to keep them around after we make the transition.

Reply:
I see. If there is no need for backwards compatibility, then no. Carry on. :)

Reply:
like having two versions of the tcf modal? one supporting vendor list v2 and one supporting v3?

Reply:
I guess? Okay, here's how it could come up (note that it requires a number of things to exist that don't exist yet): say customers could pick vendors from the vendor list. Say also that something changed between v2 and v3 that meant we couldn't simply transition their vendor list over (say, a vendor was dropped from v2 to v3). This could make them unwilling to migrate to v3 right away.
I don't think it's relevant. I don't think we're going to allow customers to be able to pick their own vendors before we roll out v3. But perhaps the thought exercise was useful.